larray_editor/editor.py

```diff
@@ -73,6 +73,9 @@ def __init__(self, parent=None, editable=False, file_menu=False, help_menu=False
         if editable:
             self.edit_undo_stack = QUndoStack(self)
 
+        self.widget_state_settings = {}
+        self.add_widget_with_state_to_restore(self.__class__.__name__, self)
```

Comment:
is self.__class__.__name__ necessary here? Won't it be added automatically in add_widget_with..., resulting in an ugly name like: "state/MappingEditor_MappingEditor"?